[Haris-Irfan] iP

Haris-Irfan opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Haris-Irfan committed:
Implement basic skeleton of chatbot

Haris-Irfan committed:
Implement switch statement and scanner in main

Haris-Irfan committed:
Rename project from Duke to KieTwoForOne

Haris-Irfan committed:
Implement adding and listing of tasks functionality

Haris-Irfan committed:
Implement Task class

Haris-Irfan committed:
Implement mark and unmark functionality

Haris-Irfan committed:
Implement event, todo, and deadline functionality

Haris-Irfan committed:
Implement automated Text UI testing

Haris-Irfan committed:
Implement exceptions

Haris-Irfan committed:
Convert Task[] array to ArrayList<Task>

Haris-Irfan committed:
Implement deletion functionality

Haris-Irfan committed:
Update automated Text UI testing

wenjebs commented on src/main/java/KieTwoForOne.java:
```diff
@@ -0,0 +1,171 @@
+        System.out.println(separationLine);
+    }
+
+    public static boolean isCompleteInput(String[] input) throws KieTwoForOneException {
+        if (input.length <= 1 && !input[0].equalsIgnoreCase("list") && !input[0]. equalsIgnoreCase("bye")) {
+            throw new KieTwoForOneException();
+        }
+        return true;
+    }
+
+    public static boolean isCompleteEventInput(String[] input) throws KieTwoForOneException {
+        if (input.length != 3) {
+            throw new KieTwoForOneException();
+        }
+        return true;
+    }
+
+    public static boolean isCompleteDeadlineInput(String[] input) throws KieTwoForOneException {
+        if (input.length != 2) {
+            throw new KieTwoForOneException();
+        }
```
Perhaps could change to 
isInputComplete, isEventInputComplete, and isDeadlineInputComplete
to sound more explicit

wenjebs commented on src/main/java/KieTwoForOne.java:
```diff
@@ -0,0 +1,171 @@
+
+public class KieTwoForOne {
+
+    private static ArrayList<Task> tasks = new ArrayList<>(100);
```
The number 100 is a magic number. 
Consider defining it as a constant with a meaningful name, such as MAX_TASKS?

wenjebs commented on src/main/java/KieTwoForOne.java:
```diff
@@ -0,0 +1,171 @@
+                    }
+                    break;
+                case TODO:
+                    KieTwoForOne.addTasks(new Todo(instruction[1]));
+                    break;
+                case EVENT:
+                    try  {
+                        isCompleteEventInput(taskDetails);
+                    } catch (KieTwoForOneException e){
+                        System.out.println("Please input a start and end time!");
+                        System.out.println(separationLine);
+                        break;
+                    }
+                    KieTwoForOne.addTasks(new Event(taskDetails[0], taskDetails[1], taskDetails[2]));
+                    break;
+                case DEADLINE:
+                    try {
+                        isCompleteDeadlineInput(taskDetails);
+                    } catch (KieTwoForOneException e) {
+                        System.out.println("Please input a deadline!");
+                        System.out.println(separationLine);
+                        break;
+                    }
+                    KieTwoForOne.addTasks(new Deadline(taskDetails[0], taskDetails[1]));
+                    break;
+                case DELETE:
+                    try {
+                        deleteTask(Integer.valueOf(instruction[1]));
+                    } catch (IndexOutOfBoundsException e) {
+                        System.out.println("Task does not exist!");
+                        System.out.println(separationLine);
+                        break;
+                    }
+                    break;
+                default:
+                    break;
+            }
+            if (instruction[0].equalsIgnoreCase("bye")) {
+                break;
+            }
```
Fix styling to align with styleguide?

Edwin1022 commented on src/main/java/Duke.java:
```diff
@@ -1,13 +1,37 @@
+
+        while(true) {
+            String instruction = scanner.nextLine();
+            switch (instruction.toUpperCase()) {
```
Shouldn't there be no indentation for `case` clauses?

Edwin1022 commented on src/main/java/KieTwoForOne.java:
```diff
@@ -0,0 +1,57 @@
+
+        while(true) {
+            String instruction = scanner.nextLine();
+            switch (instruction.toUpperCase()) {
```
Shouldn't there be no indentation for `case` clauses?

Edwin1022 commented on src/main/java/KieTwoForOne.java:
```diff
@@ -2,25 +2,44 @@
 
-    public static void addTasks(String task) {
-        tasks[count] = task;
+    public enum Instructions {
```
I like how you use `enum` to represent all the instructions.

Edwin1022 commented on src/main/java/KieTwoForOne.java:
```diff
@@ -30,24 +49,34 @@ public static void main(String[] args) {
+            String input = scanner.nextLine().toUpperCase();
+            String[] instruction = input.split(" ", 2);
+
+            switch (Instructions.valueOf(instruction[0])) {
```
Shouldn't there be no indentation for `case` clauses?

Haris-Irfan committed:
Implement saving functionality

Haris-Irfan committed:
Merge branch 'branch-level-7'

Haris-Irfan committed:
Implement date time reading and filtering functionality

Haris-Irfan committed:
Merge branch 'branch-level-8'

Haris-Irfan committed:
Implement more OOP

Haris-Irfan committed:
Merge branch 'A-MoreOOP'

Haris-Irfan committed:
Implement packages

Haris-Irfan committed:
Merge branch 'A-Packages'

Haris-Irfan committed:
Merge branch 'A-Gradle'

Haris-Irfan committed:
Implement main package kietwoforone to put all classes in

Haris-Irfan committed:
Modify build.gradle from duke to KieTwoForOne

Haris-Irfan committed:
Implement JUnit testing

Haris-Irfan committed:
Merge branch 'A-JUnit'

Haris-Irfan committed:
Configue build.gradle to create JAR file

Haris-Irfan committed:
Merge branch 'A-Jar'

Haris-Irfan committed:
Implement filter by word functionality

Haris-Irfan committed:
Merge branch 'Level-9'

Haris-Irfan committed:
Implement Java documentation

Haris-Irfan committed:
Merge branch 'A-JavaDoc'

# Conflicts:
#	src/main/java/kietwoforone/commands/FindDateCommand.java
#	src/main/java/kietwoforone/tasks/Deadline.java
#	src/main/java/kietwoforone/tasks/Task.java
#	src/main/java/kietwoforone/ui/UI.java

Haris-Irfan committed:
Add one line space between showSameDate and showMatchingTask methods

Haris-Irfan committed:
Edit code in line with coding standards

Haris-Irfan committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/kietwoforone/commands/ByeCommand.java
#	src/main/java/kietwoforone/commands/FindDateCommand.java
#	src/main/java/kietwoforone/parser/Parser.java
#	src/main/java/kietwoforone/tasks/Deadline.java

Haris-Irfan committed:
Fix issue: missing FIND case in switch statement

Haris-Irfan committed:
Separate logic statements in the parse method

Haris-Irfan committed:
Configure build.gradle to support checkstyle usage

Haris-Irfan committed:
Merge branch 'A-CheckStyle'

Haris-Irfan committed:
Implement GUI

Haris-Irfan committed:
Merge branch 'Level-10'

Haris-Irfan committed:
Update README.md

Haris-Irfan committed:
Update README.md

Haris-Irfan committed:
Update README.md

Haris-Irfan committed:
Update README.md

Haris-Irfan committed:
Update README.md

Haris-Irfan committed:
Update README.md

Haris-Irfan committed:
Update README.md

Haris-Irfan committed:
Merge branch 'master' of https://github.com/Haris-Irfan/ip

Haris-Irfan committed:
Implement chatbot closing functionality

Haris-Irfan committed:
Implement assertions

Haris-Irfan committed:
Improve code quality

Include the exception thrown in the default case of the switch statement
in the parse method. Ensures that by default any invalid inputs will
display an "Invalid input" in the chatbot.

Include missing Java Documentation for some methods

Haris-Irfan committed:
Merge pull request #2 from Haris-Irfan/branch-A-Assertions

Implement assertions

Implement assertions to ensure certain assumptions in different
methods are true.

Haris-Irfan committed:
Merge branch 'master' of https://github.com/Haris-Irfan/ip

Haris-Irfan committed:
Merge branch 'master' into branch-A-CodeQuality

Haris-Irfan committed:
Merge pull request #3 from Haris-Irfan/branch-A-CodeQuality

Improve code quality

Include the exception thrown in the default case of the switch statement
in the parse method. Ensures that by default any invalid inputs will
display an "Invalid input" in the chatbot.

Include missing Java Documentation for some methods

Haris-Irfan committed:
Merge branch 'master' of https://github.com/Haris-Irfan/ip

Haris-Irfan committed:
Implement tagging functionality

Tagging functionality allows users to tag their tasks with certain
keywords such as #work and #play to better categorise the tasks in their
list

Done by inputting tag (index) /(tag) in the chatbot

Haris-Irfan committed:
Merge branch 'branch-C-Tagging'

Haris-Irfan committed:
Finalize features.

Fixed issue with tagging functionality where invalid inputs were not being handled properly.

Fixed issue with GUI where data from the tasks.txt file was not being
loaded into the chatbot.

Fixed issue where adding a deadline did not include the time of the deadline of the task.

Haris-Irfan committed:
Added file path creation functionality in the Storage class.

Creates the file path needed to store task data when the user first opens the chatbot.

Haris-Irfan committed:
Update README.md

Included a screenshot of the application with data population (listing of different tasks currently added within the task list)

A brief introduction to the chatbot and an in-depth guide to the
different features that users can use.

Haris-Irfan committed:
Include information on bye command to exit the application

Haris-Irfan committed:
Update Ui.png

Haris-Irfan committed:
Clean up syntax in README.md

Haris-Irfan committed:
Fix tagging functionality

The tagging method only tagged the task temporarily. Needed it such that
tag was a permanent addition to the task

Haris-Irfan committed:
Fixed JavaDoc Comments.

Edited JavaDoc comments so that they are in line with coding standards.

Haris-Irfan committed:
Update README.md

Description is not accurate or appealing to users

Include more syntax styling and emojis

Include more information on features

Haris-Irfan committed:
Update README.md

User guide for downloading the application not provided in README

Include header for a guide to download the chatbot

Haris-Irfan committed:
Update README.md

User guide for downloading the application not provided in README

Include short description on where users can find the JAR file and how to start the chatbot

Haris-Irfan committed:
Update README.md

User guide for downloading the application not provided in README

Include a fenced code block to guide users on how to open JAR file

Haris-Irfan committed:
Update README.md

User guide for downloading the application not provided in README

Edit description to provide users more clarity on where they can find the release and how to run the JAR file

Haris-Irfan committed:
Update README.md

README not formatted clearly

Remove "Its features are listed below"

Following section is not features

Haris-Irfan committed:
Update README.md

README not formatted clearly

Add text for users to know where the features can be found

Helps users find the features section

Haris-Irfan committed:
Update README.md

README not visually appealling

Add italics to make README more attractive to users

Adds variance to the text

Haris-Irfan committed:
Update README.md

README not visually appealing

Include bold words to make the README more attractive to users

Adds variance to the text

Haris-Irfan committed:
Update README.md

README not formatted clearly

Add header for introduction

Helps users identify the introduction
